parity-crypto/Cargo.toml

```diff
@@ -16,13 +16,13 @@ required-features = ["publickey"]
 [dependencies]
 tiny-keccak = { version = "2.0", features = ["keccak"] }
 scrypt = { version = "0.2.0", default-features = false }
-parity-secp256k1 = { version = "0.7.0", optional = true }
+secp256k1 = { version = "0.17.2", optional = true, features = ["recovery", "rand-std"] }
 ethereum-types = { version = "0.8.0", optional = true }
 lazy_static = { version = "1.0", optional = true }
 ripemd160 = "0.8.0"
 sha2 = "0.8.0"
-digest = "0.8.1"
-hmac = "0.7.1"
+digest = "0.8.0"
+hmac = "0.7.0"
```

Comment:
is this an unintended merge resolution change?

Reply:
No, it was actually required to compile in upstream. I am still not sure exactly why but something in `parity-ethereum` has a `=0.8.0` requirement, which is crazy. Will take another look though.

Reply:
```
$ cargo check
    Updating crates.io index
error: failed to select a version for `digest`.
    ... required by package `libsecp256k1 v0.3.4 (https://github.com/paritytech/libsecp256k1.git#a922afe6)`
    ... which is depended on by `ethcore-secretstore v1.0.0 (/Users/dvd/dev/parity/parity-ethereum/secret-store)`
    ... which is depended on by `parity-ethereum v2.8.0 (/Users/dvd/dev/parity/parity-ethereum)`
versions that meet the requirements `= 0.8.0` are: 0.8.0

all possible versions conflict with previously selected packages.

  previously selected package `digest v0.8.1`
    ... which is depended on by `parity-crypto v0.4.2 (/Users/dvd/dev/parity/parity-common/parity-crypto)`
    ... which is depended on by `authority-round v0.1.0 (/Users/dvd/dev/parity/parity-ethereum/ethcore/engines/authority-round)`
    ... which is depended on by `spec v0.1.0 (/Users/dvd/dev/parity/parity-ethereum/ethcore/spec)`
    ... which is depended on by `basic-authority v0.1.0 (/Users/dvd/dev/parity/parity-ethereum/ethcore/engines/basic-authority)`
    ... which is depended on by `ethcore v1.12.0 (/Users/dvd/dev/parity/parity-ethereum/ethcore)`
    ... which is depended on by `block-gas-limit v0.1.0 (/Users/dvd/dev/parity/parity-ethereum/ethcore/block-gas-limit)`

failed to select a version for `digest` which could resolve this conflict
```

So that doesn't really make any kind of sense at all. [`libsecp256k1`](https://github.com/paritytech/libsecp256k1/blob/master/Cargo.toml#L18) asks for `digest = "0.8"`.

Reply:
I switched to specifying minor versions rather than patch versions. Not sure if it's better or worse.

Reply:
it's fine, the version conflict will probably have to be resolved in the consumer binary